test/liquid_voting/delegations_test.exs

```diff
@@ -61,24 +64,47 @@ defmodule LiquidVoting.DelegationsTest do
       assert {:ok, %Delegation{} = delegation} = Delegations.create_delegation(args)
     end
 
-    test "create_delegation/1 with proposal url sets global boolean to false", context do
-      proposal_url = "https://www.someorg/proposalX"
+    test "create_delegation/1 with duplicate data returns error changeset", context do
+      Delegations.create_delegation(context[:valid_attrs])
+      assert {:error, %Ecto.Changeset{}} = Delegations.create_delegation(context[:valid_attrs])
+    end
 
-      args = Map.merge(context[:valid_attrs], %{proposal_url: proposal_url})
-      {:ok, %Delegation{} = delegation} = Delegations.create_delegation(args)
+    test "upsert_delegation/1 with duplicate delegator and proposal_url updates the respective delegation",
+         context do
+      args = Map.merge(context[:valid_attrs], %{proposal_url: @proposal_url})
+      {:ok, %Delegation{} = original_delegation} = Delegations.create_delegation(args)
 
-      assert delegation.global == false
-    end
+      args = Map.merge(context[:update_attrs], %{proposal_url: @proposal_url})
 
-    test "create_delegation/1 without proposal url sets global boolean to true", context do
-      {:ok, %Delegation{} = delegation} = Delegations.create_delegation(context[:valid_attrs])
+      assert {:ok, %Delegation{} = modified_delegation} = Delegations.upsert_delegation(args)
+      assert original_delegation.organization_id == modified_delegation.organization_id
+      assert original_delegation.delegate_id != modified_delegation.delegate_id
+      assert original_delegation.delegator_id == modified_delegation.delegator_id
 
-      assert delegation.global == true
+      assert(
+        Delegations.list_delegations(original_delegation.organization_id)
+        |> Enum.count() ==
+          1
+      )
     end
 
-    test "create_delegation/1 with duplicate data returns error changeset", context do
-      Delegations.create_delegation(context[:valid_attrs])
-      assert {:error, %Ecto.Changeset{}} = Delegations.create_delegation(context[:valid_attrs])
+    test "upsert_delegation/1 for global delegation with duplicate delegator updates the respective delegation",
+         context do
+      {:ok, %Delegation{} = original_delegation} =
+        Delegations.create_delegation(context[:valid_attrs])
+
+      assert {:ok, %Delegation{} = modified_delegation} =
+               Delegations.upsert_delegation(context[:update_attrs])
+
+      assert original_delegation.organization_id == modified_delegation.organization_id
+      assert original_delegation.delegate_id != modified_delegation.delegate_id
+      assert original_delegation.delegator_id == modified_delegation.delegator_id
+
+      assert(
+        Delegations.list_delegations(original_delegation.organization_id)
+        |> Enum.count() ==
+          1
+      )
     end
 
```

Comment:
Would be great to have scenarios for delegation creation with upserts, as well, to complement the update scenarios above.

Reply:
I think this went unaddressed as well?

Reply:
I added:
```
test "upsert_delegation/1 with valid proposal_specific delegation data creates a delegation",
         context do
      assert {:ok, %Delegation{} = delegation} =
               Delegations.upsert_delegation(context[:valid_proposal_specific_attrs])
end

test "upsert_delegation/1 with valid global delegation data creates a delegation", context do
      assert {:ok, %Delegation{} = delegation} =
               Delegations.upsert_delegation(context[:valid_attrs])
end
```